microsoft/testsuites/dpdk/common.py

```diff
@@ -17,6 +22,249 @@
+        self.packages = packages
+        self.stop_on_match = stop_on_match
+
+
+class DependencyInstaller:
+    # provide a list of OsPackageDependencies for a project
+    def __init__(self, requirements: List[OsPackageDependencies]) -> None:
+        self.requirements = requirements
+
+    # evaluate the list of package dependencies,
+    def install_required_packages(
+        self, node: Node, extra_args: Union[List[str], None]
+    ) -> None:
+        os = node.os
+        if not isinstance(os, Posix):
+            return
+        # find the match for an OS, install the packages.
+        # stop on list end or if exclusive_match parameter is true.
+        for requirement in self.requirements:
+            if requirement.matcher(os) and requirement.packages:
+                os.install_packages(requirement.packages, extra_args=extra_args)
+                if requirement.stop_on_match:
+                    return
+        # NOTE: It is up to the caller to raise an exception on an invalid OS
+
+
+class Downloader:
+    def __init__(self, node: Node) -> None:
+        self._node = node
+
+    def download(self) -> PurePath:
+        raise NotImplementedError("Downloader not implemented.")
+
+
+class GitDownloader(Downloader):
+    _git_repo: str = ""
+    _git_ref: str = ""
+
+    def __init__(
+        self,
+        node: Node,
+        git_repo: str,
+        git_ref: str,
+    ) -> None:
+        super().__init__(node)
+        self._git_repo = git_repo
+        self._git_ref = git_ref
+
+    # checkout the git repository into the working path
+    def download(self) -> PurePath:
+        # NOTE: fail on exists is set to True.
+        # The expectation is that the parent Installer class should
+        # remove any lingering installations
+        self._asset_path = self._node.tools[Git].clone(
+            self._git_repo,
+            cwd=self._node.get_working_path(),
+            ref=self._git_ref,
+            fail_on_exists=True,
+        )
+        return self._asset_path
+
+
+# parent class for tarball source installations
+class TarDownloader(Downloader):
+    def __init__(
+        self,
+        node: Node,
+        tar_url: str,
+    ) -> None:
+        super().__init__(node)
+        self._tar_url = tar_url
+        self._is_remote_tarball = tar_url.startswith("https://")
+
+    # fetch the tarball (or copy it to the node)
+    # then extract it
+    def download(self) -> PurePath:
+        node = self._node
+        work_path = self._node.get_working_path()
+        is_tarball = False
+        for suffix in [".tar.gz", ".tar.bz2", ".tar"]:
+            if self._tar_url.endswith(suffix):
+                is_tarball = True
+                tarfile_suffix = suffix
+                break
+        assert_that(is_tarball).described_as(
+            (
+                "Source path is not a .tar[.gz|.bz2] file. "
+                f"Tar url was set to: {self._tar_url} "
+            )
+        ).is_true
```

Comment:
It should be `is_true()`. `is_true` is the function, but not called. It won't raise exception, if the condition doesn't meet.

Reply:
Surprised that passed the linter checks! Thanks.

Reply:
fixed, resolving